modules/navier_stokes/doc/content/modules/navier_stokes/pinsfv.md

```diff
@@ -0,0 +1,146 @@
+\end{equation}
+
+Solid phase energy equation, with convective heat transfer and an energy source $\dot{Q}$:
+\begin{equation}
+\dfrac{\partial (1-\epsilon) \rho c_{ps} T_s}{\partial t} &= \nabla \cdot (\kappa_s \nabla T_s) + (1-\epsilon) \alpha (T_f - T_s) + (1-\epsilon) \dot{Q}
+\end{equation}
+
+where $\rho$ is the fluid density, $\mu$ the viscosity, $c_p$ the specific heat capacity $\alpha$ the convective heat transfer coefficient.
+
+## Implementation for a straight channel
+
+We define a straight channel using a simple Cartesian mesh.
+
+!listing modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/2d-rc.i block=Mesh
+
+The non linear variables are specified among the `INSFV` and `PINSFV` sets of variables as their evaluation
+and the computation of their gradients on faces is specific to the treatment of the Navier Stokes equations.
+
+!listing modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/2d-rc.i block=Variables
+
+The porous media Navier Stokes equation are implemented by using `FVKernels` which correspond to
+each term of the equations. In the following example, the first kernel is the mass advection kernel.
+This kernel corresponds to the conservation of mass. It acts on the pressure non-linear variable which
+appears in the mass equation through the Rhie Chow interpolation of the mass fluxes on the element faces, and
+through its action on the velocity in the momentum equation.
+
+!listing modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/2d-rc.i block=FVKernels/mass
+
+Then we add the kernels acting on the 'x' component of the momentum, so essentially transcribing the first
+component of the momentum equation into MOOSE.
+
+The momentum advection term is defined by a `PINSFVMomentumAdvection` kernel. Here the porosity is constant,
+so the porosity gradient term, defined by a `PINSFVMomentumAdvectionPorosityGradient` kernel is not included.
+The momentum equation requires a velocity and an advected quantity interpolation method. This is because this
+kernel is a `FVFlux` kernel, it uses the divergence theorem to compute the divergence based on face fluxes rather
+than on the volumetric divergence value. The velocity interpolation method may be `average` or `rc` (Rhie Chow).
+The later is preferred to avoid oscillations in pressure due to the grid colocation.
+
+!listing modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/2d-rc.i block=FVKernels/u_advection
+
+The next term is the momentum diffusion. This term arises from viscous stress in the fluid. The incompressible
+approximation simplifies its treatment. This term is also evaluated using the divergence theorem.
+
+!listing modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/2d-rc.i block=FVKernels/u_viscosity
+
+The pressure gradient term in the momentum equation is expressed using a `PINSFVMomentumPressure` term.
```

Comment:
Should we recommend the `Flux` version instead? I like that one :smile:

Reply:
It may be slower ^^

Reply:
Unless it converges better...then it could be faster.